reducers/shoppingReducer.js

```diff
@@ -0,0 +1,102 @@
+import {
+  FETCH_BANNERS,
+  FETCH_CATEGORIES,
+  FETCH_PRODUCTS,
+  SET_CATEGORY,
+  SET_CART_OPEN,
+  ADD_TO_SHIPPING,
+  DELETE_FROM_SHIPPING,
+  RESET_STORE,
+  SIGN_IN,
+  SIGN_OUT,
+} from "../actions/types";
+import _ from "lodash";
+
+const shoppingReducer = (state = {}, action) => {
+  switch (action.type) {
+    case FETCH_BANNERS:
+      return { ...state, banners: action.payload };
```

Comment:
We should segregate the responsibility of this reducer and should be having one reducer per functionality as for banners we can create a bannerReducer and combine it other reducer.

similarly add to cart, remove from cart, update quantity can be moved to a cartreducer.